README.md

```diff
@@ -99,101 +140,122 @@ var excitedWords = excite(wordList);
+is visible, it becomes invisible, but if it is already invisible, it instead
+becomes visible.
+
+In the first version below, the function takes no arguments, alters the DOM
+after checking the global variable `changeTransition`, then changes the the
+global variable `changeTransition`.
 
 ```js
 var changeTransition = true;
 
-function visionChange () {
-
-  var visionimage = document.getElementById('visionimage');
+function visionChange() {
+  var visionimage = document.getElementById("visionimage");
 
   if (changeTransition === true) {
-  visionimage.style.opacity = 0;
+    visionimage.style.opacity = 0;
   }
 
   if (changeTransition === false) {
     visionimage.style.opacity = 1;
   }
 
-
   if (changeTransition === true) {
-      changeTransition = false;
+    changeTransition = false;
   } else {
-      changeTransition = true;
+    changeTransition = true;
   }
-
 }
 ```
-Next we rewrite the function as two separate functions, both of which will return the same value every time, when given the same argument.
 
-Can you see what each functions return? How might you test the functions?
+Next we rewrite the function as three separate functions, both of them will
+return the same value every time, when given the same argument.
 
+Can you see what each functions return? How might you test the functions?
 
 ```js
-function visionChange (changeTransition) {
-  if (changeTransition){
-      return false;
+function transitionToggle(changeTransition) {
+  if (changeTransition) {
+    return false;
   } else {
-      return true;
+    return true;
   }
 }
 
-function updateDom (changeTransition) {
-  return function() {
-    var visionimage = document.getElementById('visionimage');
-
-    if (changeTransition) {
-        visionimage.style.opacity = 0;
-    } else {
-        visionimage.style.opacity = 1;
-    }
+function opacityPicker(changeTransition) {
+  if (changeTransition) {
+    return 0;
+  } else {
+    return 1;
   }
 }
 
-var impureUpdateDom = updateDom(visionChange(changeTransition));
+function updateDom(newOpacity) {
+  var visionimage = document.getElementById("visionimage");
+  visionimage.style.opacity = newOpacity;
+}
+
+var impureUpdateDom = updateDom(
+  opacityPicker(transitionToggle(changeTransition))
+);
 
 // When we're ready...
 impureUpdateDom();
 ```
 
-`changeTransition()` returns either `true` or `false`, and we can test whether the function returns the expected boolean case.
+`transitionToggle()` returns either `true` or `false`, and we can test whether
+the function returns the expected boolean case.
+
+`opacityPicker()` returns either 0 or 1 , and we can test whether the function
+returns the expected number.
 
-`updateDOM` returns a function which is impure (because it changes the DOM). We can test that the returned output is indeed a function.
+`updateDOM` is a function which is impure (because it changes the DOM). and it
+is not testable because it does not return anything.
```

Comment:
This is not strictly true. `updateDOM` _is_ testable, but because it interacts with the `document` object, it is more difficult to do so, because we need to make sure we are testing it in an environment where a `document` object exists.

In week 2 we encourage people to focus on testing pure code because it is easier to get up an running with TDD without having to introduce DOM testing and it also encourages separating pure and impure code, which is often a good software design decision.